sw/device/silicon_creator/rom_ext/rom_ext_start.S

```diff
@@ -157,6 +165,27 @@ _rom_ext_start_boot:
   la   t0, (_rom_ext_interrupt_vector + 1)
   csrw mtvec, t0
 
+  /**
+   * Initialize the Ibex simple address translation to run position dependent
+   * code on the virtual flash space.
+   */
+  .extern _flash_start
+  .extern _vflash_start
+  .extern _vflash_size
+  la a0, _vflash_start
+  la a1, _vflash_size
+  la a2, _flash_start
+  call _ibex_address_translation_set  
+
+  /**
+   * Configure the epmp to enable read and execution on the virtual flash 
+   * space.
+   */
+  la a0, _vflash_start
+  la a1, _vflash_size
+  li a2, (EPMP_CFG_L | EPMP_CFG_R | EPMP_CFG_X)
```

Comment:
This is fine for a first pass but we will need to shrink the executable part so that only text sections are executable. This means we'll need to have two ePMP regions - one for read-only data and one for read-execute data. I think you're probably already aware of this since you have the necessary functions but I just wanted to make it clear.

Reply:
I did it using the Mask_rom as an example, but it needed three entries on the ePMP.
```
| Entry | Description                   | Permissions | Addressing Mode |
|-------|-------------------------------|-------------|-----------------|
| 7     | vflash `ROM_EXT` TEXT start   | RX          | TOR             |
| 8     | vflash `ROM_EXT` end          | RX          | TOR             | 
| 9     | vFlash                        | R           | NAPOT           |
```